public/javascripts/case-progress.js

```diff
@@ -143,11 +143,47 @@
 
       targetHearingIdInput.value = target.dataset.hearingid
       sendOutcomeToAdminButton.dataset.targetHearingId = target.dataset.hearingid
-      modal.style.display = "block";
+      modal.style.display = "block"
     })
   }
   // ---- Case Workflow add hearing outcome END
 
+  // ---- Assign Modal START
+  const assignOutcomeModal = document.querySelector("#assign-outcome-modal")
+
+  if(assignOutcomeModal) {
+    const assignOutcomeModalCloseButton = assignOutcomeModal.getElementsByClassName("modal-close")[0]
+    const assignOutcomeForm = assignOutcomeModal.querySelector("#assign-outcome-form-row")
+    const assignOutcomeViewLink = assignOutcomeModal.querySelector("#assign-outcome-view")
+    const targetDefendantIdInput = assignOutcomeForm.querySelector('#targetDefendantId')
+    const targetCourtCodeInput = assignOutcomeForm.querySelector('#targetCourtCode')
+
+    assignOutcomeModalCloseButton.onclick = () => {
+      assignOutcomeModal.style.display = "none"
+    }
+
+    function openAssignOutcomeModal (event) {
+      event.preventDefault()
+      const target = event.target
+
+      const viewLink = '/' + target.dataset.courtcode + '/hearing/' + target.dataset.hearingid + '/defendant/' + target.dataset.defendantid + '/summary'
+      const submitLink = '/hearing/' + target.dataset.hearingid + '/outcome/assign'
+      
+      targetDefendantIdInput.value = target.dataset.defendantid
+      targetCourtCodeInput.value = target.dataset.courtcode
+      assignOutcomeForm.setAttribute('action', submitLink)
+      assignOutcomeViewLink.href = viewLink
+      assignOutcomeModal.style.display = "block"
+    }
+
+    var assignLinks = document.getElementsByClassName('pac-assign')
```

Comment:
So rather than assigning a handler (per row I presume here), could we have a single handler and use event propagation?

Reply:
You know that might just work - let me see if I have time to get that in!